### PR TITLE
Allow basic auth with clientId

### LINKED ISSUE #74: Using a clientId should no longer be forcing token auth in the 1.1 spec
cf ably/docs#345

### PR DESCRIPTION
**Pull request by jdavid**
Also send X-Ably-ClientId (RSA7e2)

Fixes #74

Includes the commits from PR #100

### CODE REVIEW

File: src/Auth.php [hunk 1]
```diff
@@ -159,7 +158,12 @@ public function getAuthHeaders() {
             $headers[] = 'Authorization: Basic ' . base64_encode( $this->defaultAuthOptions->key );
         } else {
             $this->authorizeInternal( [], [], $force = false ); // authorize only if necessary
-            $headers[] = 'Authorization: Bearer '. base64_encode( $this->tokenDetails->token );
+            $headers[] = 'Authorization: Bearer ' . base64_encode( $this->tokenDetails->token );
+        }
+
+        // RSA7e2
+        if ( ! empty( $this->defaultAuthOptions->clientId ) ) {
+            $headers[] = 'X-Ably-ClientId: ' . base64_encode( $this->defaultAuthOptions->clientId );
```
Comment (owenpearson) [inline src/Auth.php:166]: Will this line overwrite the value of $`headers` and prevent the Authorization headers from being sent?

**Review by owenpearson — CHANGES_REQUESTED** [review]
Just one comment and one question, otherwise looks good :+1:

File: src/Auth.php (hunk 1 above)
Comment (jdavid) [reply src/Auth.php:166]: This is the PHP syntax for appending to an array:

```
$ php -a
Interactive shell

php > $headers = [];
php > $headers[] = 'Authorization: Bearer ' . 'TOKEN';
php > $headers[] = 'X-Ably-ClientId: ' . 'clientId';
php > print_r ($headers);
Array
(
    [0] => Authorization: Bearer TOKEN
    [1] => X-Ably-ClientId: clientId
)
```

Comment (owenpearson) [reply src/Auth.php:166]: oh nice, I stand corrected

**Pull request merged**